skate-plugin/src/main/kotlin/com/slack/sgp/intellij/ChangelogParser.kt

```diff
@@ -26,7 +33,9 @@ private val String.isLocalDate: Boolean
     return LOCAL_DATE_REGEX.matches(this)
   }
 
-object ChangelogParser {
+@State(name = "ChangelogParser", storages = [Storage("ChangelogParser.xml")])
+@Service(Service.Level.PROJECT)
+class ChangelogParser : PersistentStateComponent<ChangelogParser.State> {
```

Comment:
I actually linked everything up to ChangelogPresenter, do you mean to split up this and the other class or splitting ChangelogParser itself?

Reply:
Yeah I mean let's keep the original `ChangeLogParser` as it was (totally focused on parsing strings) and manage state in a separate class. Splicing them together complicates both using and testing them because they become unnecessarily coupled.

Reply:
Ah okay that makes sense, thanks for the clarification